content/content-tool/tool/src/java/org/sakaiproject/content/entityproviders/ContentEntityProvider.java

```diff
@@ -295,6 +304,50 @@ public boolean entityExists(String id) {
         }
 		return rv;
 	}
+
+	/**
+	 * return the original URL string based on Web Link resource content 
+	 * @param resource
+	 * @param contentType
+	 * @return
+	 */
+	private String getWebLinkResourceUrlString(ContentResource resource, String contentType) {
+		String urlString = null;
+		
+		// for Web Link resources
+		// return the original URL link, instead of access URL
+		if (contentType.equalsIgnoreCase(ResourceProperties.TYPE_URL))
+		{
+			try
+			{
+				byte[] content = resource.getContent();
```

Comment:
Note that this change is just for type "ResourceProperties.TYPE_URL", which is the Web Link resources. 

According to Sakai source, there is a limit to the URL content text field: 

<label for="content_0">Web Address (URL)</label>
<input type="text" name="content_0" id="content_0" size="60" maxlength="512"/>